### PR TITLE
Expand fuzz exploration envelope

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary

- replace duplicate-heavy steady-state corpus replay with unique invalid, compatible-combination, and media-source exploration
- add track coverage, cached video and malformed/path edge-case sources, plus RT-DETR, YOLO-Worldv2, and prompt-free YOLOE corpus entries (RT-DETR clamped to `imgsz=160`, the repo test convention — its 300-query decoder needs ≥160px of anchors)
- exercise CoreML on the macOS shard and increase each fuzz budget from 285 to 300 minutes
- report unique commands and skipped duplicate samples, with shell-safe reproducer commands
- keep argument canonicalization from ever touching the positional mode/task tokens (`val` is both a mode and a bool cfg key), and mark probes whose value equals the package default as effectively valid

Deleted: the steady-state `corpus` sampling strategy, duplicate effective CLI arguments, and the mode-keyed source-pool lookup in `repro` (candidates are filtered by filename, so one union of all pools replaces it).

## Validation

- `ruff format`, `ruff check`, `py_compile`, `git diff --check` on `.github/scripts/fuzz.py`
- 45-trial local smoke (`--max-trials 45`, chaos): all 35 canaries pass — including 4 track canaries and the RT-DETR/YOLO-Worldv2/YOLOE alternates — 0 timeouts/crashes/flakes, `infra_failed=False`; mutated trials exercised multi-group track combos, cached-video sources, and randomized numeric probes
- 20k-trial sampler invariant check: positional mode/task tokens always survive mutation and canonical argv never carries duplicate keys (the pre-fix head loses the `val` token at seed 221)
- `repro` replays a runner-origin `source=.../fuzz-sources/path with spaces 🚀.jpg` command by remapping to the local cached copy; `report --dry-run` confirmed stdlib-only (no `ultralytics` module imported)

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🧪 Expands Ultralytics fuzz testing to cover tracking, additional model families, realistic media sources, and more diverse command combinations.

### 📊 Key Changes
- Added `track` mode fuzzing with tracker configurations, video inputs, and tracking-specific timeouts.
- Added alternate pretrained models, including RT-DETR, YOLO-World, and YOLOE segmentation models.
- Introduced valid and malformed image, video, directory, Unicode, and space-containing file paths for source validation.
- Added safe boundary testing for resource-sensitive settings such as image size, batch size, workers, and video stride.
- Improved mutation handling to respect last-value-wins behavior, model-specific parameter floors, and supported value ranges.
- Added optional Core ML export testing when `coremltools` is available.
- Improved command quoting and reproduction support with `shlex`, including portable path remapping across operating systems.
- Prevented duplicate fuzz commands and reported unique-command and skipped-duplicate statistics.
- Increased the default fuzzing budget from 285 to 300 minutes and workflow timeouts accordingly.
- Updated CI dependencies so each runner can use the extras required for its assigned fuzzing personality.

### 🎯 Purpose & Impact
- 🛡️ Broadens automated coverage across tracking, exports, source loading, and multiple model families.
- 🐞 Improves detection of both input-validation issues and deeper runtime bugs with more realistic and malformed inputs.
- 🌍 Makes fuzz findings easier to reproduce across Linux, Windows, and macOS, including commands containing quoted or platform-specific paths.
- ⚡ Reduces wasted CI time by skipping duplicate trials and exploring more unique command combinations.
- 📈 Provides longer and more comprehensive fuzzing runs, increasing the likelihood of catching rare regressions before release.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:
- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). This is an automated review, and an engineer will assist if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The expanded corpus and shell-safe command handling are generally well structured. Two issues need attention: semantic tracking is not a valid canary input, and source-path reproduction currently references `mode` before initialization.

💬 Posted 2 inline comments

File: .github/scripts/fuzz.py [hunk 1]
```diff
@@ -215,41 +286,78 @@ def build_corpus(uni):
         # Bare weight names keep issue repro commands portable; they resolve against the precached weights_dir
         model, data = uni["task2model"][task], uni["task2data"][task]
         for mode in MODES:
+            if mode == "track" and task in {"classify", "semantic"}:
```
Comment (UltralyticsAssistant) [inline .github/scripts/fuzz.py:285]: ⚠️ **HIGH**: `semantic` also needs to be excluded from the track canaries. `SemanticSegmentationPredictor` produces `Results` with `boxes=None`, while the tracking callback unconditionally evaluates `result.boxes.cpu()` for every non-OBB task. This makes the unmutated `track semantic model=yolo26n-sem.pt ...` corpus entry fail and will repeatedly create a confirmed T1 finding rather than exercising a known-good canary.

**Suggested change:**
```suggestion
            if mode == "track" and task in {"classify", "semantic"}:
```

File: .github/scripts/fuzz.py [hunk 2]
```diff
@@ -507,9 +639,14 @@ def portable(arg):
...
-            # PureWindowsPath splits on both separators, so Windows-origin issue commands remap on any OS
-            local = (WEIGHTS_DIR if k == "model" else ASSETS) / PureWindowsPath(v).name
-            if local.exists():
+            # PureWindowsPath splits on both separators, so Windows-origin issue commands remap on any OS.
+            if k == "model":
+                candidates = [WEIGHTS_DIR / PureWindowsPath(v).name]
+            else:
+                prepare_sources(uni)
+                candidates = [ASSETS / PureWindowsPath(v).name]
+                candidates.extend(Path(p) for p, _valid in uni["sources"][mode])
```
Comment (UltralyticsAssistant) [inline .github/scripts/fuzz.py]: 💡 **MEDIUM**: `mode` is referenced here before it is assigned: the `argv = [portable(a) for a in argv]` call occurs before the mode extraction on line 654. Reproducing any command whose missing local `source=` path needs remapping therefore raises `NameError` instead of replaying the command. This affects the newly emitted cached-source repro commands on another runner or developer machine.

File: .github/scripts/fuzz.py (hunk 1 above)
Comment (glenn-jocher) [reply .github/scripts/fuzz.py:285]: Already the case on the live head: this line reads `if mode == "track" and task in {"classify", "semantic"}:` — identical to the suggestion. Verified empirically in a local 45-trial smoke at 5b8a9bb08: the corpus builds no semantic track canary (semantic entries are train/val/predict/export only) and all 4 track canaries (detect/segment/pose/obb) pass.

File: .github/scripts/fuzz.py (hunk 2 above)
Comment (glenn-jocher) [reply .github/scripts/fuzz.py]: Fixed in 5b8a9bb08 by deleting the dependency instead of reordering: `portable()` now unions all cached source pools (candidates are already filtered by exact filename match, so the mode-keyed lookup added nothing), and `mode` is extracted after remapping as on main. Verified: `repro "yolo predict detect model=yolo26n.pt (source=/home/runner/.../fuzz-sources/path with spaces 🚀.jpg) imgsz=32"` remaps to the local cached copy and passes.

**Comment by UltralyticsAssistant** [comment]
🎉 Merged—thank you, @glenn-jocher, for this excellent expansion of Ultralytics fuzz testing!

As Thomas Edison said, “The value of an idea lies in the using of it.” By applying broader, more realistic, and cross-platform fuzz coverage across tracking, exports, model families, media sources, and command combinations, this PR turns proactive testing into stronger reliability and faster discovery of regressions.

Your work helps make Ultralytics more robust for everyone. 🚀
